src/codec.cairo

```diff
@@ -82,11 +82,29 @@ pub impl EncodeOutpoint of Encode<OutPoint> {
     }
 }
 
+pub impl EncodeWitness of Encode<Span<ByteArray>> {
+    fn encode_to(self: Span<ByteArray>, ref dest: ByteArray) {
+        for witness in self {
+            dest.append(witness);
+        };
+    }
+}
+
 pub impl EncodeTx of Encode<Transaction> {
     fn encode_to(self: Transaction, ref dest: ByteArray) {
         self.version.encode_to(ref dest);
+        if (self.is_segwit) {
```

Comment:
That way you won't be able to calculate txid which does not include segwit fields (after upgrade as well) only wtxid.
Probably `is_segwit` parameter name was misleading, better use smth like `include_witness`.

Reply:
You mean all transactions after the segwit block will be is_segwit = true?
There are still many current transactions that are not yet segwit, so I don't see the point of setting it to true if it doesn't have a witness field.